From 36d415080900e558b30c312c93c5ffcaab996682 Mon Sep 17 00:00:00 2001
From: He Zhe <zhe.he@windriver.com>
Date: Tue, 3 Sep 2019 15:28:03 +0800
Subject: [PATCH 1/2] aufs: bugfix, opts: Fix missing break statement

commit 4a80018d718f6be2e862d0c70bb11cd6e6d1fae5 upstream

Add missing break statement for case Opt_wsum in au_opt_simple.

Signed-off-by: He Zhe <zhe.he@windriver.com>
See-also: https://www.mail-archive.com/aufs-users@lists.sourceforge.net/msg05684.html
(cherry picked from commit 9e7d29356ab57fa5db7e92e51267176e50c1497c)
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 fs/aufs/opts.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/aufs/opts.c b/fs/aufs/opts.c
index c92ca14574ff..a825549c029a 100644
--- a/fs/aufs/opts.c
+++ b/fs/aufs/opts.c
@@ -1386,6 +1386,7 @@ static int au_opt_simple(struct super_block *sb, struct au_opt *opt,
 	case Opt_wsum:
 		au_opt_clr(sbinfo->si_mntflags, SUM);
 		au_opt_set(sbinfo->si_mntflags, SUM_W);
+		break;
 	case Opt_nosum:
 		au_opt_clr(sbinfo->si_mntflags, SUM);
 		au_opt_clr(sbinfo->si_mntflags, SUM_W);
-- 
2.19.1

